pocket_coffea/utils/plot_utils.py

```diff
@@ -200,65 +301,149 @@ def get_systematic_uncertainty(
                 errUp = varUp.values() - nom.values()
                 errDown = varDown.values() - nom.values()
 
+                # Compute the flags to check which of the two variations (up and down) are pushing the nominal value up and down
                 up_is_up = errUp > 0
+                down_is_down = errDown < 0
+                # Compute the flag to check if the uncertainty is one-sided, i.e. when both variations are up or down
+                is_onesided = (up_is_up ^ down_is_down)
 
                 # Sum in quadrature of the systematic uncertainties
-                syst_err2_up += np.where(up_is_up, errUp**2, errDown**2)
-                syst_err2_down += np.where(up_is_up, errDown**2, errUp**2)
+                # N.B.: valid only for double sided uncertainties
+                syst_err2_up_twosided = np.where(up_is_up, errUp**2, errDown**2)
+                syst_err2_down_twosided = np.where(up_is_up, errDown**2, errUp**2)
+                syst_err2_max = np.maximum(syst_err2_up_twosided, syst_err2_down_twosided)
+                syst_err2_up_onesided = np.where(is_onesided & up_is_up, syst_err2_max, 0)
+                syst_err2_down_onesided = np.where(is_onesided & down_is_down, syst_err2_max, 0)
+                syst_err2_up_combined = np.where(is_onesided, syst_err2_up_onesided, syst_err2_up_twosided)
+                syst_err2_down_combined = np.where(is_onesided, syst_err2_down_onesided, syst_err2_down_twosided)
+
+                syst_err2_up_dict[syst] += syst_err2_up_combined
+                syst_err2_down_dict[syst] += syst_err2_down_combined
```

Comment:
Ok very nice! This should be a much more consistent implementation of prefit uncertainty band plot.